src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.cs

```diff
@@ -293,11 +277,6 @@ public static MemoryMappedFile CreateNew(string? mapName, long capacity, MemoryM
                 throw new ArgumentOutOfRangeException(nameof(access));
             }
 
-            if (access == MemoryMappedFileAccess.Write)
```

Comment:
What I mentioned above - (line 290 in this file) when we call `CreateCore`, the first argument is `null` (fileStream). This exception will not be thrown anymore with `CreateNew`.